### PR TITLE
Add Stripe billing foundation

### PR DESCRIPTION
**Pull request by BlueHotDog**
## Summary
- Add Stripe Managed Payments Checkout support for monthly/yearly subscription trials through a Req-backed billing client.
- Persist billing customers, subscriptions, and Stripe webhook events with idempotent webhook handling.
- Replace Stripe Mox coverage with PaperTiger-backed billing workflows and registered webhook delivery integration tests.
- Exercise real webhook delivery through a dynamic-port Phoenix endpoint: PaperTiger signs and posts to `/api/stripe/webhook`, the app verifies raw-body signatures, and DB records are written through the controller path.
- Keep focused Bypass/client tests for Stripe params PaperTiger does not validate, including Managed Payments params and signature edge cases.
- Add billing access states and expose billing status/allowance from `GET /api/billing/status` for client UI decisions.
- Add a Settings Billing tab that shows plan status and starts monthly/yearly Stripe Checkout when access is inactive.
- Align the billing status payload on `access_allowed` for the client schema and controller tests.
- Fix provider-settings anchor button warnings and suppress internal browser dev Sentry reporting via a Vite-defined flag.

## Verification
- `mix test test/frontman_server/billing/stripe_integration_test.exs`: 2 passed.
- Focused billing/Stripe suite passed: `mix test test/frontman_server/billing_test.exs test/frontman_server_web/controllers/billing_controller_test.exs test/frontman_server/billing/stripe_integration_test.exs test/frontman_server/billing/stripe_client_test.exs test/frontman_server/billing/subscription_test.exs`: 24 passed.
- `make precommit` passed: 891 passed, 1 skipped.
- Commit hook passed: `rescript-format`, `elixir-format-server`, `elixir-credo-server`.
- Push hook passed: `reanalyze`.
- `make rescript-build` could not run because an existing `rescript.exe watch` process holds the build lock: PID `2028366`.

### CODE REVIEW

File: .opencode/package-lock.json
Comment (itayadler) [inline .opencode/package-lock.json:1]: ?

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      on: c.id == s.billing_customer_id,
+      where: c.user_id == ^user_id
+  end
+
+  def trial_consumed(query \\ __MODULE__) do
+    from s in query,
+      where: s.status == "trialing" or not is_nil(s.trial_end)
+  end
+
+  def allow_access?(%__MODULE__{status: "trialing"}), do: true
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/subscription.ex]: hmm, they all return true / active/past_due/trialing, weird

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def for_user(query \\ __MODULE__, user_id) do
+    from s in query,
+      join: c in Customer,
+      on: c.id == s.billing_customer_id,
+      where: c.user_id == ^user_id
+  end
+
+  def trial_consumed(query \\ __MODULE__) do
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/subscription.ex]: what does it mean trial_consumed? what is this digestion?

File: .opencode/package-lock.json
Comment (BlueHotDog) [reply .opencode/package-lock.json:1]: idk why it keeps pushing this :( i think as i open opencode from this folder. i think its safe to keep but not sure

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 2 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex]: digestion?
trial_consumed means a user already used a trial so we do not offer them a trial next time they signup.
this flow is less needed now as users cannot delete their account though, just wanted to think about it.

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 1 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex]: i can simplify that, the idea is that while trialing/active/past_due we allow access, everything else we do not

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  def for_user(query \\ __MODULE__, user_id) do
+    from s in query,
+      join: c in Customer,
+      on: c.id == s.billing_customer_id,
+      where: c.user_id == ^user_id
+  end
+
+  def with_consumed_trial(query \\ __MODULE__) do
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/subscription.ex:37]: still weird naming, a trial can expire / is active or inactive

File: apps/frontman_server/lib/frontman_server/billing/webhooks.ex [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    Logger.warning(
+      "stripe webhook failed event_id=#{event_id} type=#{type} reason=#{inspect(reason)}"
+    )
+  end
+
+  defp insert_event(event_id, type, event) do
+    now = DateTime.utc_now(:second)
+
+    {count, _rows} =
+      Repo.insert_all(
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: really? this is so quirky (the 0 / 1 thingy, I guess thats how postgresql works)

File: apps/frontman_server/lib/frontman_server/billing/webhooks.ex [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  end
+
+  defp upsert_customer(attrs) do
+    %Customer{user_id: attr_value(attrs, :user_id)}
+    |> Customer.changeset(attrs)
+    |> Repo.insert(
+      on_conflict: [
+        set: [
+          stripe_customer_id: attr_value(attrs, :stripe_customer_id),
+          stripe_customer_account_id: attr_value(attrs, :stripe_customer_account_id),
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: why do we have also accoujnt_id?

File: apps/frontman_server/lib/frontman_server/billing/webhooks.ex (hunk 4 above)
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: maybe can we simplify to insert instead of insert_all

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 3 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex:37]: well almost. what happens if a user unsubscribed in the middle of a trial? in this case, next time he re-subscribed - we dont allow a trial again. but trial was never expired.

File: apps/frontman_server/lib/frontman_server/billing/webhooks.ex (hunk 4 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: done

Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: fixed. made simpler and less "smart"

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  @foreign_key_type :binary_id
+  @type t :: %__MODULE__{}
+
+  schema "billing_subscriptions" do
+    field :stripe_subscription_id, :string
+    field :stripe_customer_id, :string
+    field :stripe_customer_account_id, :string
+    field :status, :string
+    field :interval, Ecto.Enum, values: @intervals
+    field :price_id, :string
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/subscription.ex:19]: whats price_id?

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    field :current_period_end, :utc_datetime
+    field :trial_end, :utc_datetime
+    field :cancel_at, :utc_datetime
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing/subscription.ex:22]: whats the difference between cancel_at and canceled_at?

File: apps/frontman_server/lib/frontman_server/billing.ex [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+  @doc """
+  Returns whether the scoped user can receive their lifetime trial.
+  """
+  @spec trial_eligible?(Scope.t()) :: boolean()
+  def trial_eligible?(%Scope{} = scope) do
+    scope
+    |> trial_consumed_query()
+    |> Repo.exists?()
+    |> Kernel.not()
```
Comment (itayadler) [inline apps/frontman_server/lib/frontman_server/billing.ex:74]: nice

File: apps/frontman_server/test/frontman_server/billing_test.exs [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+      good_event =
+        put_in(
+          bad_event,
+          ["data", "object", "status"],
+          "trialing"
+        )
+
+      assert {:ok, :processed} = Webhooks.process_event(good_event)
+
+      assert %Subscription{stripe_subscription_id: "sub_retryable"} =
```
Comment (itayadler) [inline apps/frontman_server/test/frontman_server/billing_test.exs]: what happens if we create_customer twice? for the same user

File: apps/frontman_server/test/frontman_server/tasks_test.exs [hunk 10]
```diff
@@ -33,6 +34,43 @@ defmodule FrontmanServer.TasksTest do
     end
   end
 
+  describe "submit_user_message/5 billing access" do
```
Comment (itayadler) [inline apps/frontman_server/test/frontman_server/tasks_test.exs]: I only see here the sad path, what about successfully being able to submit a message when the user has a subscription?

File: apps/frontman_server/test/frontman_server_web/controllers/billing_controller_test.exs [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+      conn = get(conn, ~p"/api/billing/status")
+
+      assert %{
+               "status" => "trialing",
+               "interval" => "monthly",
+               "price_id" => "price_monthly_test",
+               "allow_access" => true
+             } = json_response(conn, 200)
+    end
```
Comment (itayadler) [inline apps/frontman_server/test/frontman_server_web/controllers/billing_controller_test.exs:189]: these are nice tests

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 3 above)
Comment (itayadler) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex:37]: the state machine will allow this to happen as you're saying? otherwise the state doesn't reflect that

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 6 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex:19]: coming directly from stripe. i guess if you change price you get a different ID

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 7 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex:22]: canel_at - planned date to cancel
cancled_at - when actual cancel happened

File: apps/frontman_server/lib/frontman_server/billing/subscription.ex (hunk 3 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/subscription.ex:37]: sure, why not? we dont even control that - the user can go to stripe and cancel in the middle of their trial.
and although the trial was never expired, they're not allowed to trial again

File: apps/frontman_server/lib/frontman_server/billing/webhooks.ex (hunk 5 above)
Comment (BlueHotDog) [reply apps/frontman_server/lib/frontman_server/billing/webhooks.ex]: removing
